collectors/mimecast/utils.js

```diff
@@ -30,41 +30,27 @@ function getAPILogs(authDetails, state, accumulator, maxPagesPerInvocation) {
 
                 AlLogger.debug(`MIME00009 calling url: ${url}`);
                 const tempPayload = {
+                    method: 'POST',
                     url: url,
                     headers: requestHeaders,
-                    body: JSON.stringify(applicationDetails.payload)
+                    data: applicationDetails.payload,
                 };
-                const payloadData = applicationDetails.encoding === null ||  applicationDetails.encoding ? { ...tempPayload, encoding: applicationDetails.encoding } : tempPayload;
-                request.post(payloadData, function (error, response, body) {
-                    if (error) {
-                        return reject(error);
-                    }
-                    if (response.statusCode &&
-                        (response.statusCode === 429 ||
-                            response.statusCode >= 500)) {
-                        return reject(response);
-                    }
-
+                const payloadData = applicationDetails.encoding === null || applicationDetails.encoding ? { ...tempPayload, responseType: 'arraybuffer', encoding: applicationDetails.encoding } : tempPayload;
+                axios.request(payloadData).then(response => {
                     if (!applicationDetails.compress) {
-                        try {
-                            body = JSON.parse(body);
-                        } catch (exception) {
-                            AlLogger.error("MIME000010 Error parsing response. ", body);
```

Comment:
hope this is working fine with normal logs and logs in zip file? can you check when collector is deployed or in local

Reply:
yes. I have checked locally. for  SiemLogs we get the zip file then we have method unzipBufferInMemory  and for other  streams we don’t need the parse after axios api changes